zarnitsa/series/DataAugmenterTimeSeries.py

```diff
@@ -0,0 +1,159 @@
+
+    def augment_column(
+        self,
+        col: pd.Series,
+        func: str,
+        **kwargs,
+    ) -> pd.Series:
+        """Augment dataframe data. Pandas dataframe"""
+        func = {
+            "dropping": self.data_droping,
+            "data_interpolation": self.data_interpolation,
+            "data_extrapolation": self.data_extrapolation,
+            "data_denoiser": self.data_denoiser,
+            "data_noiser": self.data_noiser,
+        }
+        return func[func_name](col, **kwargs)
+
+    def _prepare_data_to_aug(self, data, freq=0.2) -> Tuple[pd.Series, pd.Series]:
+        pass
+
+    @staticmethod
+    def data_droping(col, limit=None, freq=0.2, **kwargs) -> pd.Series:
+        """
+        Extrapolate pandas column
+        """
+        if len(col) == 0:
+            raise ValueError(f"Iterable object <{type(col)}> is empty! Check input!")
+        if limit:
+            freq = limit / col.shape[0]
+        drop_indices = col.sample(frac=freq).index
+        return col[[i for i in col.index if i not in drop_indices]]
+
+    @staticmethod
+    def data_interpolation(
+        col, method="linear", limit=None, freq=1.0, order=2, **kwargs
+    ) -> pd.Series:
+        """
+        Interpolate pandas column
+        filling the gaps
+        """
+        if len(col) == 0:
+            raise ValueError(f"Iterable object <{type(col)}> is empty! Check input!")
+
+        if limit is None:
+            limit = int(col[col.isna()].shape[0] * freq)
+
+        if method in ["fillna"]:
+            col = col.fillna(kwargs["value_to_fill"], limit=limit)
+        elif method in ["linear", "pad", "polynomial"]:
+            indices = col.index
+            col = col.reset_index(drop=True)
+            col = col.interpolate(method=method, limit=limit, **kwargs)
+            col.index = indices
+        elif method in ["polyfit"]:
+            non_missed, missed = col[~col.isna()], col[col.isna()]
+            poly = np.poly1d(np.polyfit(non_missed.index, non_missed.values, order))
+            na_records = missed.sample(frac=limit / missed.shape[0])
+            col.loc[na_records.index] = poly(na_records.index)
+        else:
+            raise KeyError(f"Undefined method <{method}>!")
+
+        return col
+
+    @staticmethod
+    def data_extrapolation(
+        col, method="polyfit", limit=None, freq=0.2, h=None, **kwargs
+    ) -> pd.Series:
+        """
+        Extrapolate pandas column
+        """
+        if len(col) == 0:
+            raise ValueError(f"Iterable object <{type(col)}> is empty! Check input!")
+        if limit is None:
+            limit = int(col[col.isna()].shape[0] * freq)
+
+        if h is None:
+            h = (max(col.index) - min(col.index)) / col.shape[0]
+
+        if method in ["polyfit"]:
+            poly = np.poly1d(np.polyfit(col.index, col.values, kwargs["order"]))
+            indices_to_extrapolate = [
+                col.index[-1] + (h * (t + 1)) for t in range(limit)
```

Comment:
seems like a magic, consider adding comments to things like this please

Reply:
here we assume that the index defines time steps. there can be some gaps in data, btw. So, we define the "time" step size and add N extra indices for the column data extrapolation